app/models/poll.rb

```diff
@@ -1,21 +1,21 @@
 # frozen_string_literal: true
 class Poll < ActiveRecord::Base
   require 'vote-schulze'
+  require 'poll_status_machine'
+  include PollStatusMachine
 
-  enum status: {
-    draft:     'draft',
-    ready:     'ready',
-    published: 'published',
-    closed:    'closed',
-    deleted:   'deleted'
-  }
+  AVAILIBLE_TRANSITIONS = { draft:     { 'ready' => 'draft', 'finished' => 'draft', 'deleted' => 'draft' },
```

Comment:
I'd really prefer this not to be a global frozen constant — this hash is only needed once, when the class is loading. There is completely no sense to keep it in memory for the whole lifecycle of a program.

Also, while it is expected for states to be _adjectives_ (`ready`, `finished` suit here just fine) — but the transitions themselves are better to be _verbs_ (`finish`, like a command, not  `finished` as a state). This is. of course, is just an English language semantics, and doesn't affect validity of your code anyhow.